Makefile

```diff
@@ -205,14 +209,19 @@ deploy-service-requirements: deploy-namespace deploy-inventory-service-file
 		--ocp-versions '$(OPENSHIFT_VERSIONS)' --ocp-override "$(OPENSHIFT_INSTALL_RELEASE_IMAGE)" --public-registries "$(PUBLIC_CONTAINER_REGISTRIES)" \
 		$(E2E_TESTS_CONFIG)
 
-deploy-service: deploy-namespace deploy-service-requirements deploy-role
+deploy-resources: manifests
+	python3 ./tools/deploy_crd.py $(ENABLE_KUBE_API_CMD)
+
+deploy-service: deploy-namespace deploy-service-requirements deploy-role deploy-resources
 	python3 ./tools/deploy_assisted_installer.py $(DEPLOY_TAG_OPTION) --namespace "$(NAMESPACE)" \
-		--profile "$(PROFILE)" $(TEST_FLAGS) --target "$(TARGET)" --replicas-count $(REPLICAS_COUNT)
+		--profile "$(PROFILE)" $(TEST_FLAGS) --target "$(TARGET)" --replicas-count $(REPLICAS_COUNT) \
+		$(ENABLE_KUBE_API_CMD)
 	python3 ./tools/wait_for_assisted_service.py --target $(TARGET) --namespace "$(NAMESPACE)" \
 		--profile "$(PROFILE)" --domain "$(INGRESS_DOMAIN)"
 
-deploy-role: deploy-namespace
-	python3 ./tools/deploy_role.py --namespace "$(NAMESPACE)" --profile "$(PROFILE)" --target "$(TARGET)"
+deploy-role: deploy-namespace manifests
```

Comment:
it means that we are going for solution of generating the roles upon deployment and not during build, right?

Reply:
Yes, i can add it the build as well but it will not affect anything related to the code. 
`generate-from-swagger` is not part of the build target as well.